pkg/cmd/cli/secrets/source.go

```diff
@@ -0,0 +1,213 @@
+	CreateSourceSecretRecommendedCommandName = "new-gitconfig"
+
+	createSourceSecretLong = `
+Create a new source secret
+
+Source secrets are used to authenticate against Git server.
+`
+)
+
+type CreateSourceSecretOptions struct {
+	SecretNamespace string
+	SecretName      string
+	Username        string
+	Password        string
+	Token           string
+
+	GitConfigPath   string
+	CertificatePath string
+
+	SecretsInterface client.SecretsInterface
+
+	Out io.Writer
+}
+
+func NewCmdCreateSourceSecret(name, fullName string, f *cmdutil.Factory, out io.Writer, newSecretFullName, ocEditFullName string) *cobra.Command {
+	o := &CreateSourceSecretOptions{Out: out}
+
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s SECRET_NAME --username=USERNAME --password=PASSWORD --token=TOKEN --cert-path=CERTIFICATE", name),
+		Short: "Create a new source secret",
+		Long:  fmt.Sprintf(createSourceSecretLong, fullName, newSecretFullName, ocEditFullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := o.Complete(f, args); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.Validate(); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.CreateSourceSecret(); err != nil {
+				cmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Username, "username", "", "username for Git authentication")
+	cmd.Flags().StringVar(&o.Password, "password", "", "password for Git authentication")
+	cmd.Flags().StringVar(&o.Token, "token", "", "token for Git authentication")
+	cmd.Flags().StringVar(&o.CertificatePath, "cert-path", "", "path to the certificate file")
+	// cmd.Flags().StringVar(&o.GitConfigPath, "gitconfig-path", "", "path to the gitconfig file")
+	cmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+func (o *CreateSourceSecretOptions) CreateSourceSecret() error {
+	secret, err := o.MakeSourceSecret()
+	if err != nil {
+		return err
+	}
+
+	if _, err := o.SecretsInterface.Create(secret); err != nil {
+		return err
+	}
+
+	// fmt.Fprintf(o.GetOut(), "secret/%s\n", secret.Name)
+
+	return nil
+}
+
+type SourceConfigEntry struct {
+	// Username    string
+	Password    string
+	Token       string
+	Certificate string
+}
+
+const (
+	GitConfigKey = ".gitconfig"
+)
+
+func (o *CreateSourceSecretOptions) MakeSourceSecret() (*api.Secret, error) {
+
+	ca_content, err := ioutil.ReadFile(o.CertificatePath)
+	if err != nil {
+		return nil, err
+	}
+
+	gitAuth := SourceConfigEntry{
+		// Username:    o.Username,
+		Password:    o.Password,
+		Token:       o.Token,
+		Certificate: string(ca_content),
+	}
+
+	gitConfig := map[string]SourceConfigEntry{o.Username: gitAuth}
+
+	gitConfigContent, err := json.Marshal(gitConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	secret := &api.Secret{}
+	secret.Namespace = o.SecretNamespace
+	secret.Name = o.SecretName
+	secret.Type = api.SecretTypeOpaque
+	secret.Data = map[string][]byte{}
+	secret.Data[GitConfigKey] = gitConfigContent
```

Comment:
Is this format actually compatible with `~/.gitconfig`?  Having the names the same, but the content different seems a little confusing.